dallinger/command_line.py

```diff
@@ -313,71 +319,64 @@ def debug(verbose, bot):
                 if verbose:
                     sys.stdout.write(line)
 
                 # Open browser for new participants
-                match = re.search('New participant requested: (.*)$', line)
-                if match:
-                    url = match.group(1)
-                    if bot:
-                        log("Using a bot to simulate participant...")
-                        try:
-                            from dallinger_experiment import Bot
-                            participant = Bot(url)
-                            participant.run_experiment()
-                        except ImportError:
-                            log("This experiment does not have a bot.")
-                    else:
+                if not bot:
+                    match = re.search('New participant requested: (.*)$', line)
+                    if match:
+                        url = match.group(1)
                         webbrowser.open(url, new=1, autoraise=True)
 
-                # Is recruitment over? We can end this debug session.
+                # Is recruitment over? We can end the loop, and check
+                # experiment status
                 match = re.search('Close recruitment.$', line)
                 if match:
-                    if participant:
-                        # make sure there are no stray phantomjs processes
-                        participant.driver.quit()
                     log('Recruitment is complete.')
                     break
 
-                # Check for bot exceptions
-                match = re.search('Exception on ', line)
-                if match and not verbose:
-                    error("There was an error running the experiment.")
-                    if participant:
-                        participant.driver.quit()
+            # Check status url
+            status_url = base_url + '/summary'
+            while True:
+                time.sleep(30)
+                data = {}
+                try:
+                    resp = requests.get(status_url)
+                    data = resp.json()
+                except (ValueError, requests.exceptions.RequestException):
+                    error('Error fetching experiment status.')
+                log('Current application state: {}'.format(data))
+                if data.get('completed', False):
+                    log('Experiment completed, all nodes filled.')
                     break
 
-                # Check for unexpected bot hangs
-                match = re.search('Ignoring EPIPE', line)
-                if participant and match:
-                    epipe = epipe + 1
-                    if epipe >= 2:
-                        error("The experiment finished but recruitment was not closed.")
-                        participant.driver.quit()
-                        break
-
     finally:
-        p.kill()
+        try:
+            # It seems we need to explicitly kill all subprocesses with a SIGINT
```

Comment:
I believe the `CTRL_C_EVENT` is supposed to (as should psutil), though I'm also not sure if any of this heroku local stuff will work on Windows.